pfcpiface/notifier_test.go

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022-present Open Networking Foundation
+
+package pfcpiface
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_downlinkDataNotifier_Notify(t *testing.T) {
+	n := &downlinkDataNotifier{
+		notifyChan:           make(chan<- uint64, 1024),
+		notificationInterval: 5 * time.Second,
+	}
+
+	testFSEID := uint64(0x1)
+
+	n.Notify(testFSEID)
+	require.Len(t, n.notifyChan, 1)
+	n.Notify(testFSEID)
+	// we haven't picked any event from channel, so length should be the same.
+	require.Len(t, n.notifyChan, 1)
+}
+
+func Test_downlinkDataNotifier_shouldNotify(t *testing.T) {
```

Comment:
nit: this is testing the internal API (`shouldNotify`), but not the public one, i.e. that messages are actually sent into the channel.

Reply:
Could you clarify? Indeed, the goal of `Test_downlinkDataNotifier_shouldNotify` is to test the internal API. The public one is tested by `Test_downlinkDataNotifier_Notify`.